File: 12-week-9--searching-and-sorting-continued/02-day-5--find-value-binary-tree/javascript/find_target.js
class Node {
  constructor(value, left = null, right = null) {
    this.value = value;
    this.left = left;
    this.right = right;
  }
}

function findTarget(root, target) {
  // type your code here
}

if (require.main === module) {
  // add your own tests in here
  const root = new Node(1, new Node(-1), new Node(2));
  console.log("Expecting: Node with value 2");
  console.log(findTarget(root, 2));

  console.log("");

  console.log("Expecting: null");
  console.log(findTarget(root, 5));
}

module.exports = { findTarget, Node };

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 12-week-9--searching-and-sorting-continued/02-day-5--find-value-binary-tree/ruby/find_target.rb
class Node
  attr_accessor :value, :left, :right

  def initialize(value, left = nil, right = nil)
    @value = value
    @left = left
    @right = right
  end
end

def find_target(root, target)
  # type your code in here
end

if __FILE__ == $PROGRAM_NAME
  root = Node.new(1, Node.new(-1), Node.new(2))
  puts "Expecting: Node with value 2"
  puts find_target(root, 2).inspect

  puts

  puts "Expecting: nil"
  puts find_target(root, 5).inspect

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
